### DISCUSSION TITLE
Indent new line like previous line

### DISCUSSION THREAD

**Discussion by SpecificProtagonist** [helix-editor/helix#4532] [Q&A]
How can I tell Helix to indent new lines (when I press enter) the same amount as previous lines? Currently it seemingly only indents as much as the previous line has full tabs, ignoring any additional spaces. For example, I have
```
foo
- bar
- baz
    - long text
      that takes multiple lines
```

and want to add another line to the last item, I press enter, but the new line only has 4 spaces, not 6.

**Comment by kj** [comment]
I would like to see something like this as I've been trying out [Nickel](https://nickel-lang.org/) where it seems common to write `let` bindings at the top of a file with the same indentation, e.g.

```
let foo = "foo" in
let bar = "bar" in
let baz = "baz" in

{
  # ...
}
```

The language indentation expects you to indent each level of `let` so for the rest of the file (which could get quite long) the editor automatically adds three levels of indentation every time you start a new line. For example, say you have written up to the first `{` above, and then start a new line. What you get is:

```
{
        _
      }
```

-where `_` is the cursor position. That continues for every line thereafter, no matter how far you nest records, you'll always have three levels extra indentation on the next line which you need to unindent, which gets pretty tedious!

So either Helix can somehow 'reset' the indentation level at some point (or recognise where you have manually adjusted the indentation level, e.g. pulling each `let` back one level, and continue from there) or a configuration option could be added at the language level to use a more simple/dumb indent style, which always indents new lines to the same indentation level as the previous line, e.g.

```
[[language]]
name = "nickel"
indent = { tab-width = 2, unit = " ", simple = true }
```
